### PR TITLE
Modified Configuration File

### PR DESCRIPTION
**Pull request by BrandonJLazard**
Added conditionals so that the user has the correct compiler and libraries configured before the actual configuration is run.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -838,9 +854,9 @@ echo ""
 # Standard LIBRARIES
 if [[ $OSTYPE == *"darwin"* ]]
 then
-    STD_FLAGS=""
+    STD_FLAGS="-lc++"
 else
-    STD_FLAGS=""
+    STD_FLAGS="-lstdc++"
```
Comment (gassmoeller) [inline configure:859]: Did you add the STD_FLAGS back in on purpose?

**Review by gassmoeller — COMMENTED** [review]
Makes sense, just one question.

File: configure (hunk 1 above)
Comment (BrandonJLazard) [reply configure:859]: I did not, not sure why those appeared back up. Will change that

**Comment by gassmoeller** [comment]
The testers dont like it: https://github.com/geodynamics/Rayleigh/actions/runs/9586908067/job/26435825639?pr=535
Have you tested that this works for you?

**Review by gassmoeller — APPROVED** [review]
I think this is now good to go.

**Pull request merged**